cargo-dist/src/init.rs

```diff
@@ -84,15 +130,117 @@ pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
         }
     };
 
+    // If we're migrating, the configuration will be missing the
+    // generic workspace specification, and will have some
+    // extraneous cargo-specific stuff that we don't want.
+    let mut workspace_toml = if is_migrating {
+        // Fetch just the fields we want
+        let dist = workspace_toml
+            .get("workspace")
+            .and_then(|t| t.get("metadata"))
+            .and_then(|t| t.get("dist"))
+            .unwrap();
```

Comment:
I don't think this can/should be an unwrap. I think the migrate code can run on a non-initted workspace rn?

Reply:
Oh yes, sorry, missed that from debugging. Will clean up.